python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -290,8 +291,10 @@ def _create_repository_using_definitions_args(
     )
     def created_repo():
         return [
-            *with_resources(assets or [], resource_defs),
-            *with_resources(asset_checks or [], resource_defs),
+            *with_resources(dedupe_objects_ref_equality(assets) if assets else [], resource_defs),
+            *with_resources(
+                dedupe_objects_ref_equality(asset_checks) if asset_checks else [], resource_defs
+            ),
             *(schedules_with_resources),
             *(sensors_with_resources),
             *(jobs_with_resources),
```

Comment:
why not schedules, sensors, and jobs as well?

Reply:
I think the underlying problem is most poignant for definitions loaded using load_... methods, which to my knowledge isn't currently possible with those categories. But happy to add them if we think it's valuable